### DISCUSSION TITLE
Handling drafts

### DISCUSSION THREAD

**Discussion by TechWatching** [statiqdev/Discussions#96] [Q&A] [answered]
How do you handle drafts on statiq.web ?

What I would need as someone writing articles on my blog is:
- Have a way (metadata for instance) to distinguish draft articles from other
- Be able to preview my draft articles when I am working on an article not yet finished (so while debugging locally)
- Prevent draft articles to be publicly available when I deploy me blog on Netlify

Is there something built-in to do that ?

I think that some static generators use a property to mark some of their posts as "draft" and then you can use their CLI to display or not drafts.

I have seem that on your personal blog @daveaglick  you use a draft folder but I don't really understand if it's just to store some draft articles that you can't preview or if there is a mechanism that allow you to preview them locally but prevent them from being published when you deploy you static site.

**Comment by daveaglick** [comment, marked as answer]
This is a great question! Folks have come up with a few way of doing this. The most common (probably because it was supported back in Wyam) is to set `Published` to a future date:

```
Published: 1/1/9999
```

That will cause the post to be excluded from the build and then you can clear or set the `Published` value when you're ready to post.

Another approach is to set `Excluded` which is new to Statiq Web. By default, `Excluded` looks at `Published` to determine if the document should be excluded. But you can configure it to something else. For example, if you want `Excluded` to look for a `Draft` front matter value instead you can set the following in a `_directory.yaml` [file](https://www.statiq.dev/web/content-and-data/metadata/directory-metadata):

```
Excluded: => Draft
```

Alternatively, you could exclude everything in an entire folder by adding a `_directory.yaml` file just to that folder:

```
Excluded: true
```

You could even get really fancy and toggle exclusion based on some other condition like a setting when working locally or not:

```
Excluded: => GetBool("SomeKey") && GetDateTime("Published") > DateTime.Today.AddDays(1)
```
